cmd/register/files.go

```diff
@@ -73,27 +88,50 @@ Override Output location prefix during registration.
 ::
 
  bin/flytectl register file  _pb_output/* -d development  -p flytesnacks --continueOnError -v v2 -l "s3://dummy/prefix"
-
+	
 Usage
 `
+	sourceCodeExtension = ".tar.gz"
 )
 
 func registerFromFilesFunc(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
 	return Register(ctx, args, cmdCtx)
 }
 
 func Register(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
-	dataRefs, tmpDir, _err := getSortedFileList(ctx, args)
-	if _err != nil {
-		logger.Errorf(ctx, "error while un-archiving files in tmp dir due to %v", _err)
-		return _err
-	}
-	logger.Infof(ctx, "Parsing files... Total(%v)", len(dataRefs))
-	fastFail := !rconfig.DefaultFilesConfig.ContinueOnError
+	var _err error
+	var dataRefs []string
+	var validProto []string
+	var tmpDir string
 	var registerResults []Result
-	for i := 0; i < len(dataRefs) && !(fastFail && _err != nil); i++ {
-		registerResults, _err = registerFile(ctx, dataRefs[i], registerResults, cmdCtx)
+
+	// getSerializeOutputFiles will return you all proto and  source code compress file in sorted order
+	dataRefs, tmpDir, err := getSerializeOutputFiles(ctx, args)
+	if err != nil {
+		logger.Errorf(ctx, "error while un-archiving files in tmp dir due to %v", err)
+		return err
+	}
+	logger.Infof(ctx, "Parsing file... Total(%v)", len(dataRefs))
+
+	for i := 0; i < len(dataRefs); i++ {
+		if len(rconfig.DefaultFilesConfig.AdditionalDistributionDir) > 0 && len(rconfig.DefaultFilesConfig.DestinationDir) > 0 && strings.HasSuffix(dataRefs[i], sourceCodeExtension) {
+			if err = uploadFastRegisterArtifact(ctx, dataRefs[i], rconfig.DefaultFilesConfig.AdditionalDistributionDir, rconfig.DefaultFilesConfig.Version); err != nil {
+				return fmt.Errorf("please check your Storage Config. It while while uploading the source code. %v", err)
+			}
+		} else if (len(rconfig.DefaultFilesConfig.AdditionalDistributionDir) == 0 || len(rconfig.DefaultFilesConfig.DestinationDir) == 0) && strings.HasSuffix(dataRefs[i], sourceCodeExtension) {
+			return errors.New("Fast serialize detected without additional flags")
+		} else if strings.HasSuffix(dataRefs[i], ".pb") {
+			validProto = append(validProto, dataRefs[i])
+		} else {
+			return errors.New("Invalid files")
+		}
 	}
+
+	fastFail := rconfig.DefaultFilesConfig.ContinueOnError
+	for i := 0; i < len(validProto) && !(fastFail && _err != nil); i++ {
+		registerResults, _err = registerFile(ctx, validProto[i], registerResults, cmdCtx)
+	}
```

Comment:
Can we do this logic.
Fix the messages and the api function appropriately as this just a draft logic which i am thinking we should use

```
1] sortedFileList = getSortedFileList(args)
2] fastSourceArchive, filteredSortedList = filter(sortedFileList, fast_register_pattern_file)
3] if fastSourceArchive != nill {
      if areFastRegisterFlagsPassed()  {
         uploadArchive(....)
      } else { 
       return err ("fast serialized source file detected in the archive, please pass the other two args")
    }
4] Regular register of filteredSortedList
```

Reply:
done